tensorboard/backend/security_validator_test.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for `tensorboard.backend.security_validator`."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+try:
+  # python version >= 3.3
+  from unittest import mock    # pylint: disable=g-import-not-at-top
+except ImportError:
+  import mock    # pylint: disable=g-import-not-at-top,unused-import
+
+import werkzeug
+from werkzeug import test as werkzeug_test
+from werkzeug.datastructures import Headers
+
+from tensorboard import test as tb_test
+from tensorboard.backend import security_validator
+from tensorboard.util import tb_logging
+
+logger = tb_logging.get_logger()
+
+_WARN_PREFIX = "In 3.0, this warning will become an error:\n"
+
+
+def create_headers(
+    content_type="application/json",
+    x_content_type_options="nosniff",
+    content_secrurity_policy=""
+):
+  return Headers(
+      {
+          "Content-Type": content_type,
+          "X-Content-Type-Options": x_content_type_options,
+          "Content-Security-Policy": content_secrurity_policy,
+      }
+  )
+
+
+class SecurityValidatorMiddlewareTest(tb_test.TestCase):
+  """Tests for `SecurityValidatorMiddleware`."""
+
+  def setUp(self):
+    super(SecurityValidatorMiddlewareTest, self).setUp()
+    self.get_header = lambda req: {}
+    app = werkzeug.Request.application(
+        lambda req: werkzeug.Response("OK", headers=self.get_header(req))
+    )
+    app = security_validator.SecurityValidatorMiddleware(app)
+    self.server = werkzeug_test.Client(app, werkzeug.BaseResponse)
+    self.mock_warn = mock.patch.object(logger, "warn").start()
+
+  def tearDown(self):
+    super(SecurityValidatorMiddlewareTest, self).tearDown()
+    mock.patch.stopall()
+
+  def make_request_and_maybe_assert_warn(
+      self, headers, expected_warn_substr=None
+  ):
+    self.get_header = lambda req: headers
+    self.server.get("")
+
+    if expected_warn_substr is None:
+      self.mock_warn.assert_not_called()
```

Comment:
If we're going to assert this on the same mock we probably should reset the mock at the top of this function, so that we don't have an ordering dependency between calls to this helper (e.g. right now wouldn't the test fail if somebody re-ordered the assertions so that the failing assertion was first in the test method?).

That said IMO it would be better to just use a fresh mock each time for isolation (versus resetting the mock over and over), and to not patch the logger (since the `mock_warn.assert_not_called()` would fail if we were to start emitting an unrelated warning from some other dependency) but instead to just mock the call to `_maybe_raise_value_error()` since we already have that separated out.  So we could do something like:

```python
with mock.patch.object(security_validator, _maybe_raise_value_error) as mock_error:
  self.server.get("")
mock_error.assert_...
```